xarray/tests/test_dataset.py

```diff
@@ -6689,17 +6689,45 @@ def test_polyfit_warnings(self) -> None:
             ds.var1.polyfit("dim2", 10, full=True)
             assert len(ws) == 1
 
-    def test_pad(self) -> None:
+    @pytest.mark.parametrize(
```

Comment:
Does this work if you want to pad along a dimension coordinate (aka. a variable that is called the same as it's dimension)?

Reply:
Good question, I'll investigate :D

Reply:
So, as I understand it most dims in the test dataset are dimension coordinates and they pad correctly, so I think so. In any case I've extended the tests to pad all dimensions just to be sure nothing is behaving incorrectly. Let me know if I missed any case.